### PR TITLE
Mock Toast Messages in tests

### PR DESCRIPTION
**Pull request by HaGuesto**
to reduce the false negatives in the FE tests.

@vahidbazzaz @pylipp

### CODE REVIEW

File: react/src/views/Transfers/ShipmentView/components/ShipmentTabs.tsx [hunk 1]
```diff
@@ -35,7 +35,7 @@ function ShipmentTabs({ shipmentDetail }: IShipmentTabsProps) {
       </TabList>
       <TabPanels>
         <TabPanel p={0}>
-          {(shipmentDetail.length || 0) === 0 && (
+          {(shipmentDetail?.length || 0) === 0 && (
```
Comment (HaGuesto) [inline react/src/views/Transfers/ShipmentView/components/ShipmentTabs.tsx:38]: @vahidbazzaz please ask chatgpt to fix this or assure that shipmentDetail is not undefined before.

File: react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx [hunk 2]
```diff
@@ -103,7 +103,7 @@ function ShipmentCard({ shipment }: IShipmentProps) {
         <StackDivider borderColor="blackAlpha.800" marginTop={-1.5} />
         <Box p={4}>
           <Center alignContent="stretch">
-            <Text fontWeight="bold">TOTAL:</Text> {shipment.details.length} boxes
+            <Text fontWeight="bold">TOTAL:</Text> {shipment.details?.length!} boxes
```
Comment (HaGuesto) [inline react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:106]: @vahidbazzaz test 4.5.1.4 was failing due to a typeError here.

**Comment by HaGuesto** [comment]
@vahidbazzaz test 4.5.1 is still failing, but not due to flaky tests. please run the test and fix them. 

To me it feels like you have missed some data in the mock for the shipment.

**Comment by HaGuesto** [comment]
[Mock toast messages for flaky FE tests](https://trello.com/c/OLxp2ynZ/1159-mock-toast-messages-for-flaky-fe-tests)

File: react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx (hunk 2 above)
Comment (vahidbazzaz) [reply react/src/views/Transfers/ShipmentView/components/ShipmentCard.tsx:106]: @HaGuesto I tested it locally, and the mock data seemed to work fine

**Comment by HaGuesto** [comment]
TODO:
merge master into it after #672 and #689 are merged to master. in the best case this resolves the failing tests.
Otherwise, @vahidbazzaz needs to take another look into his tests

**Comment by HaGuesto** [comment]
@vahidbazzaz can u merge master into this branch and resolve the conflicts. All conflicts are with the shipment PRs from you and I am not sure how to resolve them.

**Review by vahidbazzaz — APPROVED** [review]
Nice job Hans!

**Review by pylipp — APPROVED** [review]

**Pull request merged**